src/project_management.py

```diff
@@ -0,0 +1,277 @@
+import os
+from src.json_io import load_data,save_data,PROJECTS_DIR,default_file
+from datetime import datetime
+import json
+
+ARCH_DIR="archives"
+
+def list_projects(file_name: str) -> None:
+    """Lists all projects from the JSON file.
+
+    Args:
+        file_name (str): Name of the JSON file containing projects.
+    """
+
+    data = load_data(file_name)
+    
+    for i, project in enumerate(data.get("projects", [])):
+        print(f"{i+1}. {project['project_name']} - {project['description']} - Manager: {project['manager']} - Deadline: {project['deadline']}")
+
+
+def add_project(file_name: str) -> None:
+    """Adds a new project to the JSON file.
+
+    Args:
+        file_name (str): Name of the JSON file to add the project to.
+    """
+
+    data = load_data(file_name)
+    
+    while True:
+        project_name = input("Project name: ")
+        
+        
+        for project in data["projects"]:
+            if project["project_name"] == project_name:
+                print(f"A project with name '{project_name}' already exists. Please choose another name.")
+                break
+        else:
+            
+            break
+    
+    description = input("Description: ")
+    manager = input("Project manager name: ")
+    deadline = input("Deadline (YYYY-MM-DD): ")
+
+    new_project = {
+        "project_name": project_name,
+        "description": description,
+        "manager": manager,
+        "deadline": deadline,
+        "tasks": []
+    }
+    
+    data["projects"].append(new_project)
+    save_data(data, file_name)
+
+def select_project(file_name: str) -> dict[str, dict] | None:
+    """Selects a project from the list to perform actions on.
+
+    Args:
+        file_name (str): Name of the JSON file containing projects.
+
+    Returns:
+        Optional[dict]: The selected project dictionary or None if selection is invalid.
+    """
+
+    data = load_data(file_name)
+
+    list_projects(file_name)
+
+    selected_project = input("Enter the project number: ")
+
+    try:
+        selected_project_index = int(selected_project) - 1
+        project = data['projects'][selected_project_index]
+        return project
+
+    except (ValueError, IndexError):
+        print(f"Invalid project number '{selected_project}'. Please choose a valid project number.")
+        return None
+
+def list_tasks(file_name: str) -> str| None:
+    """Lists tasks for the selected project.
+
+    Args:
+        file_name (str): Name of the JSON file containing projects.
+
+    Returns:
+        str | None: The name of the selected project or None if project selection fails.
```

Comment:
Milyen selected project? Nem kap semmilyen kiválasztott projektet. Itt kerül kiválasztásra a projekt, tehát a fv. azt csinálja, hogy kiválasztatja a projektet, kilistázza a feladatait, majd visszaadja a kiválasztott projekt nevét.
Ez így nem jó, mert a projekt-kiválasztás és a feladat-listázás két különálló, független funkció, ráadásul előbbire már van egy másik függvény.
A feladat-listázáshoz a választott projekt a szükséges bemenet, tehát azt paraméterben kéne megkapnia:
```suggestion
def list_tasks(file_name: str, project: dict[str, dict]) -> None:
    """Lists tasks for the selected project.

    Args:
        file_name (str): Name of the JSON file containing projects.
        project (dict[str, dict]): The project selected for listing its tasks.
```